torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -1531,7 +1535,7 @@ c10::optional<std::unique_ptr<SchedulerEntry>> SegmentedGroup::
   FUSER_PERF_SCOPE("SegmentedGroup::getMaybeSchedulerEntry");
   auto fusion = segmented_fusion_->completeFusion();
   auto data_cache = segmented_fusion_->getCachedHeuristicDataFor(this);
-  FusionSegmentGuard fsg(fusion, runtime_inputs_, runtime_outputs_);
+  FusionSegmentGuard fsg(fusion, runtime_inputs_, runtime_outputs_, false);
```

Comment:
Why necessary to skip setting the memory types of input/output tensors?

Reply:
It saves dynamic casts to `TensorView` type since the inputs/outputs are stored as `Val*`. Also at this point the fusion has already been lowered and compiled to kernel so the memory type really wouldn't need to be updated.

Reply:
This PR is largely trying to explore low-latency without losing much readability/maintainability. So request for re-evaluation of any parts welcome  : ).

Reply:
Ah, OK. Not against this change, but does the cost of dynamic_cast really matter?

Reply:
It was just done along with optimizing `FusionSegmentGuard`, which did turn out to be bottleneck at one point. But I just checked back without skipping setting memory type specifically and the diff looked close to noise level (1~2us with google benchmark). 

Another thing I noticed is that dynamic casts and call stacks tend to be exaggerated disproportionately by profilers so these actually look quite heavy on the profiled results.